client/reader/discover/index.web.js

```diff
@@ -10,13 +9,12 @@ import { setLocaleMiddleware } from 'calypso/controller/shared';
 import { sidebar, updateLastRoute } from 'calypso/reader/controller';
 import { discover } from './controller';
 
-export default function () {
-	const langParam = getLanguageRouteParam();
+export default function ( router ) {
 	const anyLangParam = getAnyLanguageRouteParam();
-	page( `/${ anyLangParam }/discover`, redirectInvalidLanguage );
 
-	page(
-		[ '/discover', `/${ langParam }/discover` ],
+	router(
+		[ '/discover', `/${ anyLangParam }/discover` ],
```

Comment:
It's a bit unfortunate that this jumps so much when testing with a logged-in user 😕 


https://github.com/user-attachments/assets/17fef59e-98aa-45a3-ae11-4abd9070a612

Reply:
Thanks for this, I agree. I assumed the jump to and from the logged-out layout only occurred locally, but I now see it happens on calypso.live too, and it's probably caused by the changes in this PR. That's not how I intended it so that should be fixed!

Reply:
I believe calypso.live instances have different behavior when it comes to setting the logged-in state on the server side compared to production/staging running on `wordpress.com`.

When booting Calypso locally with `wpcom-user-bootstrap` enabled, the SSR flashing no longer happens, which makes me believe that this issue won't occur on production.

Reply:
It appears that we have the same SSR flashing on the `/themes` section in calypso.live and on production, it skips the SSR completely as intended.

Reply:
Thanks for checking, @yuliyan. That makes sense. I just tried running it locally with `wpcom-user-bootstrap`, but I still see the flashing. From a quick look at the code, I think I need a `wordpress_logged_in_cookie` in my `config/secrets.json`, which I set, but no luck yet. Maybe I'm still not configured correctly.

Reply:
FWIW, @yuliyan is right, I had forgotten about this behavior. 

You won't be able to test it locally, because SSR isn't disabled locally. 

You can verify this with `/themes` and `/plugins` on staging/production - it won't flash there because the user bootstrap works correctly and SSR is disabled. 

Feel free to disregard my initial comment, nothing to do here 👍